### PR TITLE
Agios 387

### PR DESCRIPTION
**Pull request by corinnekrych**
@cvasilak @danbev small refactor as discussed in 
https://github.com/cvasilak/aerogear-ios-sync/commit/274c2c18fd10ad6b58231795db8495a1aea44424#commitcomment-9774009

### CODE REVIEW

**Comment by danbev** [comment]
I was hoping for something that would clarify the original code which was hard to follow and hence maintain. But I'm finding this solution to be hard to follow as well. Perhaps I'm not the best person to review this and I'll leave it to @cvasilak to have his say. If you guys think this is better then we should go with this PR.

**Comment by corinnekrych** [comment]
@danbev instead of a for loop i'm using reduce function ie: more functional programming side.
To simplify we could extract the imbricated reduce in an external function.
I'd like it better that way...
@cvasilak wdyt?

**Comment by corinnekrych** [comment]
@danbev also I fixed one bug here: https://github.com/aerogear/aerogear-ios-sync/blob/master/AeroGearSync-DiffMatchPatch/DiffMatchPatchMessage.swift#L56

Where it was assumed there will be on;y one edit in the edits array. (not dealing with ",")

**Comment by danbev** [comment]
@corinnekrych I was hoping for a solution that used some JSON construct or dictionary to build up the content and then have it serialized to a String.

**Comment by cvasilak** [comment]
@corinnekrych nice use of reduce here! 

I think, per @danbev comment if we can also remove the string building and use a dictionary, fill it with the values and then use the ```asJsonString()``` to return it will be nice addition I guess too.

**Comment by corinnekrych** [comment]
@danbev for JSON representation, let's wait until we implement Swift version of JsonPatch lib as it will be needed there, we can then refactor and reuse.
Not sure about what you suggest @cvasilak, sort gist will be more concrete...

**Comment by danbev** [comment]
@corinnekrych I have no problem with the usage of reduce.

**Comment by corinnekrych** [comment]
@cvasilak @danbev 
Going from object model to dictionary is not that easy (dynamic vs strongly type lg is not Swift best strength):
- One way to go would be to use jsonsz and then asJsonString() as @cvasilak suggested
https://github.com/aerogear/aerogear-ios-jsonsz/blob/master/AeroGearJsonSZTests%2FAeroGearJsonSZTests.swift#L327
but might be a bit overkill to add a dependency to ag-ios-jsonsz just for that...
- we go as proposed in this PR with reduce
- or we stick to existing and close PR (but still a small bug here: https://github.com/aerogear/aerogear-ios-sync/blob/master/AeroGearSync-DiffMatchPatch/DiffMatchPatchMessage.swift#L56 when there are several edits, to be fixed)

Choose the way you want to go or another proposal (with some code) if you have more idea.

**Comment by danbev** [comment]
I've run out of time to be able to provide anything in the way of code. I think the most important thing right now is that the bug @corinnekrych found is fixed. It does not matter to me which of the solutions we choose as I don't see that as the real issue. So you an @cvasilak choose what you like better and lets take on the other task for alpha.2 or some later version.

**Comment by cvasilak** [comment]
@corinnekrych @danbev opened PR#https://github.com/corinnekrych/aerogear-ios-sync/pull/3 with what was my original thinking on it on the json gen,

**Comment by corinnekrych** [comment]
@cvasilak fine for me, it has a more OO look and might read better
+ to merge BUT only concern is on unit test which needs to be rewritten because a dictionary does not guarantee the order of its element so the string might be different on TRavis build

**Comment by danbev** [comment]
+1 I think this is easier to understand and maintain.

**Comment by corinnekrych** [comment]
@cvasilak let me do the same for DiffMatchPatchMessage + unit tests

**Comment by abstractj** [comment]
@corinnekrych would be nice if we could rebase, instead of hit the merge button from GH

**Comment by abstractj** [comment]
@corinnekrych would be nice if we could rebase, instead of hit the merge button from GH

**Comment by corinnekrych** [comment]
@cvasilak and as expected Travis get  different order ;) i'll fix that

**Comment by corinnekrych** [comment]
landed 2981d289ba3d0494140e6dc4a4a7e1a32194bf4e
thanks @cvasilak
